src/SilicaVault.ts

```diff
@@ -0,0 +1,29 @@
+import {
+    Epoch,
+    Epoch as EpochEntity,
+    SilicaVault as SilicaVaultEntity
+} from "../generated/schema";
+  
+import { StartNextRound } from "../generated/SilicaVault/SilicaVault";
+
+export function handleStartNextRound(event: StartNextRound): void {
+    let silicaVault = SilicaVaultEntity.load(event.address.toHex());
+    if (!silicaVault) {
+        silicaVault = new SilicaVaultEntity(event.address.toHex());
+        silicaVault.save();
+    }
+
+    let epochEntry = new Epoch(event.params.currentRound.toString());
+    epochEntry.startTimestamp = event.block.timestamp;
+    epochEntry.roundEnd = event.params.roundEndDay;
+    epochEntry.roundSize = event.params.roundSize;
+    epochEntry.deposited = event.params.deposited;
+    epochEntry.sharePriceAtStart = event.params.sharePriceAtStart;
+    epochEntry.shareSupplyAtStart = event.params.shareSupplyAtStart;
```

Comment:
I think the next three fields should go on the previous epoch entry (`event.params.currentRound - 1`), since the event looks like:

`pTokenWithdrawnPerRound[nextRound - 1],
            rTokenWithdrawnPerRound[nextRound - 1],
            sharesWithdrawnPerRound[nextRound - 1]`

which reminds me, i should have commented in the other PR, but we should also move those updates to the `ProcessWithdraws` event so we can get them sooner (so the UI user can finish withdraw sooner). that actually ends up being cleaner too since during processWithdraws it's `currentRound` and not `currentRound - 1`, meaning we don't have to check whether `currentRound - 1` exists in thegraph

Reply:
Ack.

Blocked on event changes in subgraph.